src/lib.rs

```diff
@@ -243,6 +246,25 @@ fn get_num_cpus() -> usize {
     cpus as usize
 }
 
+
+#[cfg(target_os = "macos")]
+fn get_num_physical_cpus() -> usize {
+    let mut cpus: i32 = 0;
+    let mut cpus_size = std::mem::size_of_val(&cpus);
+
+    let sysctl_name = CString::new("hw.physicalcpu").unwrap();
```

Comment:
For some reason the `sysctlbyname?  signature wants a `[i8]` pointer, not `[u8]` (which is produced by the b"" literal). Forcing a cast and relying on the manually added `\0` feels worse than a stack-allocated String for each call.